libs/base/Data/Either.idr

```diff
@@ -31,6 +31,52 @@ isRight : Either a b -> Bool
 isRight (Left _)  = False
 isRight (Right _) = True
 
+--------------------------------------------------------------------------------
+-- `Either` used to a instance of interface `Foldable`.
```

Comment:
You should implement `Foldable (Either e)`, not copy/paste the list of
definitions. Cf. the existing `Foldable` implementations in the prelude.

Reply:
`Foldable (Either e)` is already defined in `Prelude.Types`.
There is no need to add all of these definitions again.
There is no need to turn on all of these `%allow_overloads`.